boa/contracts/vyper/vyper_contract.py

```diff
@@ -499,16 +397,18 @@ def __init__(
 
         # set external methods as class attributes:
         self._ctor = None
+        self._address = None
         if "__init__" in external_fns:
             self._ctor = VyperFunction(external_fns.pop("__init__"), self)
 
         if skip_initcode:
             if value:
                 raise Exception("nonzero value but initcode is being skipped")
-            addr = Address(override_address)
+            self._address = Address(override_address)
         else:
-            addr = self._run_init(*args, value=value, override_address=override_address)
-        self._address = addr
```

Comment:
not sure why this was removed, it makes the behavior at the join point clear

Reply:
I didn't realize the attribute was defined in the base class. Will revert

Btw this is why i lke ternaries in general